commands.js

```diff
@@ -1044,16 +1044,48 @@ exports.commands = {
 			move = Commands.randommove.call(this, "2,viable", self, dummyRoom);
 		}
 
+		//Code for generating a random name based on types generated
+		//Start by accessing the array from the capnamelist.js file in the data folder.
+		var foundName = false;
+		var t1Name = '';
+		var type1array = Data.Capnames[type[0]];
+		var type2array = Data.Capnames[type[1]];
+		while(!foundName){
+            t1Name = Tools.sample(type1array)[0]
+            if(t1Name.charAt(0) === '-'){ //Case Suffix, can't be first part of name. remove and reroll
+                Tools.removeElm(type1array, t1Name);
```

Comment:
While I appreciate that you're using my built in tool functions, this is actually not what you want. Since `type1array` is a pointer to the base data file, you actually end up deleting (permanently for all recalls of this command) the name that did not match, which is definitely not what we want. 

A better approach would be to use `Tools#shuffle` instead of `Tools#sample`. This way, instead of using a `while` loop and the `foundName` variable, you can just iterate over the shuffled name list until you find a name that works. This also insures that you only check each name one time, and further makes sure that an infinity loops isn't possible, since even if no name is found the list will eventually be exhausted and the loop will stop.